test/test_cuda.py

```diff
@@ -1729,6 +1736,62 @@ def accum_hook(grad):
 
         self.assertEqual(x.grad, torch.ones_like(x) * 5)
 
+    # Skip the test for ROCm as per https://github.com/pytorch/pytorch/issues/53190
+    @skipIfRocm
```

Comment:
What about a simple test where a new stream is used, an operation is performed, backward happens, and the grads are checked on that same stream?

```
with stream s0:
  do some ops on a tensor
  backward
  check the grad and it's correct
```

Reply:
Added slightly more challenging version
```python
with s0:
    do some ops on a tensor
with s1:
    sync s1 with s0
    backward
    check the grad and it's correct
```
to [test_streaming_backwards_sync](https://github.com/pytorch/pytorch/pull/54227/files#diff-893b1eea27352f336f4cd832919e48d721e4e90186e63400b8596db6b82e7450R1699-R1713) above.